src/Wig.Zstd/Program.cs

```diff
@@ -29,36 +35,143 @@ public sealed class DefaultCommand : AsyncCommand<DefaultCommand.Settings>
 
         public sealed class Settings : CommandSettings
         {
-            [CommandArgument(0, "[EXAMPLE]")]
-            [Description("The example to run.\nIf none is specified, all examples will be listed")]
-            public string Name { get; set; }
+            [CommandArgument(0, "[path]")]
+            [Description("Path to the files")]
+            public string Path { get; set; }
 
-            [CommandOption("-l|--list")]
-            [Description("Lists all available examples")]
-            public bool List { get; set; }
+            [CommandOption("-c|--compress")]
+            [Description("Compress files")]
+            public bool IsCompressionMode { get; set; }
 
-            [CommandOption("-a|--all")]
-            [Description("Runs all available examples")]
-            public bool All { get; set; }
+            [CommandOption("-d|--decompress")]
+            [Description("Decompress files")]
+            public bool IsDecompressionMode { get; set; }
 
-            [CommandOption("-s|--source")]
-            [Description("Show example source code")]
-            public bool Source { get; set; }
+            [CommandOption("-o|--overwrite")]
+            [Description("Overwrite file / directory")]
+            public bool Overwrite { get; set; }
 
-            [CommandOption("--select")]
-            [Description("Show example source code")]
-            public bool Select { get; set; }
+            [CommandOption("-l|--level")]
+            [Description("Compression level (1-22)")]
+            public int CompressionLevel { get; set; }
+
+            [CommandOption("-s|--subfolder")]
+            [Description("Subfolder name")]
+            public string Subfolder{ get; set; }
         }
 
         public DefaultCommand(IAnsiConsole console)
         {
             _console = console;
         }
 
+        public async Task WriteFileAsync(byte[] data, string path, string subfolder, string extension)
+        {
+            var fileName = Path.GetFileName(path);
+            var currentDirectory = Path.GetDirectoryName(path);
+
+            var directory = Directory.CreateDirectory($"{currentDirectory}/{subfolder}");
+
+            await using (FileStream fstream = new FileStream($"{directory}/{fileName}{extension}", FileMode.OpenOrCreate))
+            {
+                await fstream.WriteAsync(data, 0, data.Length);
+            }
+        }
+
+        public async Task CompressingAsync(string path, int compressionLevel, bool overwrite, string subfolder)
+        {
+            using var options = new CompressionOptions(compressionLevel);
+            using var compressor = new Compressor(options);
+            FileAttributes attr = File.GetAttributes(path);
+            if (attr.HasFlag(FileAttributes.Directory))
+            {
+                string[] filePaths = Directory.GetFiles(path);
```

Comment:
Many unexpected things can potentially happen here. 
@mkhomutov 
Should we use orc.filesystem instead here or at least handle common possible exception from this method (like PathTooLongException etc)